person.rb

```diff
@@ -1,26 +1,31 @@
-# Person class
-class Person
-  attr_reader :id, :age
-  attr_accessor :name
+# Define the base Decorator class that inherits from Nameable
+require_relative 'nameable'
+class Person < Nameable
+  attr_accessor :id, :name, :age, :parent_permission
 
-  def initialize(age, parent_permission: true, name: 'Unknown')
+  def initialize(age, name, parent_permission: true)
+    super()
     @id = generate_id
-    @name = name
+    @name = name || 'Unknown'
     @age = age
     @parent_permission = parent_permission
   end
 
-  def can_use_service?
-    is_of_age || @parent_permission
+  def can_use_services?
+    of_age? || @parent_permission
   end
 
   def of_age?
     @age >= 18
   end
 
+  def correct_name
+    name
+  end
+
   private
 
   def generate_id
-    Random.rand(1..1000)
+    rand(1..1000)
   end
 end
```

Comment:
- In your person.rb for the following code to work , 
```
person = Person.new(22, 'maximilianus')
p  person.correct_name
 capitalizedPerson = CapitalizeDecorator.new(person)
 p capitalizedPerson.correct_name
 capitalizedTrimmedPerson = TrimmerDecorator.new(capitalizedPerson)
p  capitalizedTrimmedPerson.correct_name
```

You have to require the following classes in your person.rb:

```
require_relative 'capitalize_decorator'
require_relative 'trimer_decorator'
```

Please update it 🌞